juicebox_cli/cli.py

```diff
@@ -67,18 +56,17 @@ def login(ctx, username, env):
 @click.option('--netrc', default=None)
 @click.option('--job')
 @click.option('--app', default=None)
-@click.option('--env', envvar='JB_ENV', default='prod')
+@click.option('--endpoint', required=True)
 @click.option('--client', default=None)
 @click.pass_context
-def upload(ctx, client, env, app, job, netrc, files):
-    validate_environment(ctx, env)
-    logger.debug('Starting upload for %s - %s: %s', env, job, files)
+def upload(ctx, client, endpoint, app, job, netrc, files):
```

Comment:
It occurs to me that we should no longer require the `client` parameter to this operation: now that users are specific to a client, and we have the endpoint, we should get rid of the `--client` parameter here. This also means the `clients_list` operation should no longer be necessary and could be deleted.

I guess there are two ways to do this:

1. make it so that `juice upload` looks up the client list (and assumes there should only be a single result) and always uses that. (this only requires changes to juicebox-cli)

2. make it so the `client` is no longer passed to juicebox-public-api's `/upload-token/` endpoint at all, since it already has the `endpoint` which uniquely identifies the client. (this will require a change to juicebox-public-api).

Reply:
looks like juicebox-public-api already handles the case of no client being passed. So I think we can safely just get rid of `--client` and `clients_list`.

Reply:
Yea, I can remove the `--client` option and the `clients_list` command since I think it was there to able to supply a value to the `--client` option.